driver-async/src/main/com/mongodb/async/client/FindIterableImpl.java

```diff
@@ -47,13 +47,12 @@
 
     FindIterableImpl(final ClientSession clientSession, final MongoNamespace namespace, final Class<TDocument> documentClass,
                      final Class<TResult> resultClass, final CodecRegistry codecRegistry, final ReadPreference readPreference,
-                     final ReadConcern readConcern, final AsyncOperationExecutor executor, final Bson filter,
-                     final FindOptions findOptions) {
+                     final ReadConcern readConcern, final AsyncOperationExecutor executor, final Bson filter) {
         super(clientSession, executor, readConcern, readPreference);
         this.operations = new AsyncOperations<TDocument>(namespace, documentClass, readPreference, codecRegistry, readConcern);
         this.resultClass = notNull("resultClass", resultClass);
         this.filter = notNull("filter", filter);
-        this.findOptions = notNull("findOptions", findOptions);
+        this.findOptions = new FindOptions();
```

Comment:
There is no need to pass in FindOptions, it only risks mutation outside this class.